### ISSUE TITLE
Ruby macos builds are failing continuously on master due to gpg keyserver issues

### ISSUE THREAD

**Issue by apolcyn** [grpc/grpc#26522]
```
++ export 'RUN_TESTS_FLAGS=--filter_pr_tests --base_branch origin/master -f basictests macos ruby --internal_ci -j 1 --inner_jobs 4 --max_time=3600'
++ RUN_TESTS_FLAGS='--filter_pr_tests --base_branch origin/master -f basictests macos ruby --internal_ci -j 1 --inner_jobs 4 --max_time=3600'
++ '[' true == true ']'
++ gpg_recv_keys_success=0
++ (( i=0 ))
++ (( i<5 ))
+++ host pool.sks-keyservers.net
+++ grep 'has address'
+++ head -n1
+++ awk '{print $4}'
++ KEYSERVER_IPV4_ADDRESS=
++ gpg --keyserver hkp:// --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB
gpg: could not parse keyserver URL
++ [[ 0 == 1 ]]
++ sleep 3
++ (( i++ ))
++ (( i<5 ))
+++ host pool.sks-keyservers.net
+++ grep 'has address'
+++ awk '{print $4}'
+++ head -n1
++ KEYSERVER_IPV4_ADDRESS=
++ gpg --keyserver hkp:// --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB
gpg: could not parse keyserver URL
```

**Comment by apolcyn** [comment]
The cause of the issue right now is:

```
$ host pool.sks-keyservers.net
Host pool.sks-keyservers.net not found: 3(NXDOMAIN)
```

Not yet clear if this is a transient outage or a breaking change.

**Comment by jtattermusch** [comment]
Looks like this is indeed release blocker since it causes other problems (e.g. https://github.com/grpc/grpc/issues/26546)

**Comment by apolcyn** [comment]
Internal tracker of this issue b/191857029

**Comment by markdroth** [comment]
Should be fixed by #26565.
